matrix/lambdas/daemons/worker.py

```diff
@@ -1,21 +1,89 @@
     The worker (third) task in a distributed filter merge job.
     """
-    def __init__(self):
+    def __init__(self, request_id):
         self.lambda_handler = LambdaHandler()
         self.dynamo_handler = DynamoHandler()
+        self._request_id = request_id
+        self._deployment_stage = os.environ['DEPLOYMENT_STAGE']
+        self._bundle_uuid = None
+        self._bundle_version = None
+        self._input_start_row = None
+        self._input_end_row = None
+        self._num_rows = None
+
+    def run(self, format: str, worker_chunk_spec: dict):
+        """Process and write one chunk of dss bundle matrix to s3 and
+        invoke reducer lambda when last worker job is completed.
 
-    def run(self, request_id: str, filter_string: str, worker_chunk_spec: dict):
+        Inputs:
+        format: (str) Expected file format of request
+        worker_chunk_spec: (dict) Information about input bundle and matrix row indices to process
         """
-        Filter one work chunk and invoke reducer lambda when last worker job is completed.
+        self._parse_worker_chunk_spec(worker_chunk_spec)
+        dss_zarr_store = DSSZarrStore(bundle_uuid=self._bundle_uuid,
+                                      bundle_version=self._bundle_version,
+                                      dss_instance=self._deployment_stage)
+        group = zarr.group(store=dss_zarr_store)
+        exp_df, qc_df = convert_dss_zarr_root_to_subset_pandas_dfs(group, self._input_start_row, self._input_end_row)
+        s3_zarr_store = S3ZarrStore(self._request_id)
+        s3_zarr_store.write_from_pandas_dfs(exp_df, qc_df, self._num_rows)
+
+        field_value = StateTableField.COMPLETED_WORKER_EXECUTIONS.value
+        self.dynamo_handler.increment_table_field(DynamoTable.STATE_TABLE, self._request_id, field_value, 1)
+
+        workers_and_mappers_are_complete = self._check_if_all_workers_and_mappers_for_request_are_complete(
+            self._request_id)
+        if workers_and_mappers_are_complete:
+            s3_zarr_store.write_column_data(group)
+            reducer_payload = {
+                "request_id": self._request_id,
+                "format": format
+            }
+            self.lambda_handler.invoke(LambdaName.REDUCER, reducer_payload)
+
+    def _parse_worker_chunk_spec(self, worker_chunk_spec: dict):
+        """Parse worker chunk spec into Worker instance variables.
+
+        Input:
+            worker_chunk_spec: (dict) keys expected 'bundle_uuid', 'bundle_version',
+                               'start_row', 'num_rows'
+        """
+        self._bundle_uuid = worker_chunk_spec['bundle_uuid']
+        self._bundle_version = worker_chunk_spec['bundle_version']
+        self._input_start_row = worker_chunk_spec['start_row']
+        self._num_rows = worker_chunk_spec['num_rows']
+        self._input_end_row = self._input_start_row + self._num_rows
+
+    def _check_if_all_workers_and_mappers_for_request_are_complete(self, request_id):
```

Comment:
Prefer simpler function names. I don't think function names need to explain what the function is doing. Suggestions:
- `_is_reducer_ready`
- `_is_last_worker`